### PR TITLE
IB fuzz tests

### LINKED ISSUE #11: Generative tests for data received from IB API
IB's Python API is not well-typed or well-documented, so there will likely be tons of corner cases in our parsing code. We should have a set of fuzz tests that yield complete crap, just to make sure that our parsing will handle whatever gets thrown at it. (We can use `hypothesis` for this, like our property-based tests.)

### PR DESCRIPTION
**Pull request by jspahrsummers**
Resolves #11.

- [x] Populate expiration date for futures
- [x] Tests for position parsing

### CODE REVIEW

File: ibkr.py [hunk 1]
```diff
@@ -12,6 +12,15 @@
 import re
 
 
+def parseFiniteDecimal(input: str) -> Decimal:
+    with localcontext(ctx=Context(traps=[DivisionByZero, Overflow])):
+        value = Decimal(input)
+        if not value.is_finite():
+            raise ValueError('Input is not numeric: {}'.format(input))
```
Comment (kastiglione) [inline ibkr.py]: Mentioning just in case: Python 3.6 added string interpolation, this can be `f'Input is not numeric: {input}'`.

File: ibkr.py [hunk 2]
```diff
@@ -12,6 +12,15 @@
 import re
 
 
+def parseFiniteDecimal(input: str) -> Decimal:
+    with localcontext(ctx=Context(traps=[DivisionByZero, Overflow])):
```
Comment (kastiglione) [inline ibkr.py]: TIL about `localcontext`

File: ibkr.py (hunk 1 above)
Comment (jspahrsummers) [reply ibkr.py]: Yeah, I learned that, but only after writing a bunch of these—and then I didn't want the style to diverge. 😅 

Want to PR a codemod? :wink:

**Comment by jspahrsummers** [comment]
Hopefully these won't be too flaky! 🙏
